src/pages/RepoPage/context.tsx

```diff
@@ -0,0 +1,53 @@
+import { createContext, useContext, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+
+type Breadcrumb = {
+  pageName: string
+  text?: string
+  children?: JSX.Element
+  readOnly?: boolean
+}
+
+const RepoBreadcrumbContext = createContext<{
+  setBaseCrumbs: (crumbs: Breadcrumb[]) => void
+  breadcrumbs: Breadcrumb[]
+  setBreadcrumbs: (crumbs: Breadcrumb[]) => void
+}>({
+  setBaseCrumbs: () => {},
+  breadcrumbs: [],
+  setBreadcrumbs: () => {},
+})
+
+RepoBreadcrumbContext.displayName = 'RepoBreadcrumbContext'
+
+export const RepoBreadcrumbProvider: React.FC<React.PropsWithChildren> = ({
+  children,
+}) => {
+  const { owner } = useParams<{ owner: string }>()
+
+  const [baseCrumbs, setBaseCrumbs] = useState<Breadcrumb[]>([
+    { pageName: 'owner', text: owner },
+  ])
+  const [crumbTail, setCrumbTail] = useState<Breadcrumb[]>([])
+
+  const breadcrumbs = useMemo(
+    () => [...baseCrumbs, ...crumbTail],
+    [baseCrumbs, crumbTail]
+  )
+
+  return (
+    <RepoBreadcrumbContext.Provider
+      value={{
+        setBaseCrumbs,
+        breadcrumbs,
+        setBreadcrumbs: setCrumbTail,
+      }}
+    >
+      {children}
+    </RepoBreadcrumbContext.Provider>
+  )
+}
+
+export function useCrumbs() {
+  return useContext(RepoBreadcrumbContext)
+}
```

Comment:
Can we add a check in here to ensure that the context is actually being provided? You can see an example of that in  [here](https://github.com/codecov/gazebo/blob/main/src/ui/TopBanner/TopBanner.tsx#L58-L70) tho it is a bit more complex with the use of a zod schema, you can probably skip that part, but throwing an error if it doesn't exist is a nice way of ensuring that people don't misuse this.